src/contentnormalizer.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/contentnormalizer
+ */
+
+import UpcastWriter from '@ckeditor/ckeditor5-engine/src/view/upcastwriter';
+
+/**
+ * Content Normalizer class provides a mechanism to transform input data send through
+ * an {@link module:clipboard/clipboard~Clipboard#event:inputTransformation inputTransformation event}. It fixes an input content,
+ * which has a source in applications like: MS Word, Google Docs, etc. These applications generate content which frequently
+ * is an invalid HTML. Content normalizers transform it, what later might be properly upcast to {@link module:engine/model/model~Model}.
+ *
+ * Content Normalizers are registered by {@link module:paste-from-office/pastefromoffice~PasteFromOffice} plugin. Each instance is
+ * initialized with an activation trigger. Activation trigger is a function which gets content of `text/html` dataTransfer (String) and
+ * returns `true` or `false`. Based on this result normalizer applies filters to given data.
+ *
+ * {@link module:paste-from-office/contentnormalizer~FilterFunction Filters} are function, which are run sequentially, as they were added.
+ * Each filter gets data transformed by the previous one.
+ *
+ * Example definition:
+ *
+ * 	const normalizer = new ContentNormalizer( contentHtml =>
+ * 		contentHtml.includes( 'docs-internal-guid' )
+ * 	);
+ *
+ * 	normalizer.addFilter( ( { data, documentFragment, writer } ) => {
+ * 		// filter's content, which transforms data in clipboard
+ * 	} );
+ *
+ * Normalizers are stored inside Paste from Office plugin and are run on
+ * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation inputTransformation event}. Below example is simplified and show
+ * how to call normalizer directly on clipboard event, what has happen inside
+ * {@link module:paste-from-office/pastefromoffice~PasteFromOffice} plugin.
+ *
+ * 	editor.plugins.get( 'Clipboard' ).on( 'inputTransformation', ( evt, data ) => {
+ * 		normalizer.transform( data );
+ * 	} );
+ *
+ * @class
+ */
+export default class ContentNormalizer {
+	/**
+	 * Initialize Content Normalizer.
+	 *
+	 * @param {Function} activationTrigger The function which checks for what content should be applied this normalizer.
+	 * It takes an HTML string from the `text/html` dataTransfer as an argument and have to return a boolean value
+	 */
+	constructor( activationTrigger ) {
+		/**
+		 * Keeps a reference to the activation trigger function. The function is used to check if current Content Normalizer instance
+		 * should be applied for given input data. Check is made during the {@link #transform}.
+		 *
+		 * @private
+		 * @type {Function}
+		 */
+		this._activationTrigger = activationTrigger;
```

Comment:
This should be a simple method in a class that implements the normalizer and not passed external behavior.